### DISCUSSION TITLE
How to block local proxy ports but allow other ports

### DISCUSSION THREAD

**Discussion by EsuRt** [tnodir/fort#350] [Q&A] [answered]
I have installed proxy software on my system, with local proxy ports set to 10800 and 10801. I want to prevent software from accessing the internet through these proxy ports but still allow them to access other local ports. I set up rules in Fort to achieve this, but it doesn't work, other local ports are also being blocked. Could there be an issue with the rules configuration?

![1](https://github.com/user-attachments/assets/789c6b81-986e-4c36-a99b-877e37b4320a)

![2](https://github.com/user-attachments/assets/637bd5ef-ae1d-4db6-a263-17ea823d4637)

![3](https://github.com/user-attachments/assets/c2d55b16-2f00-4dd5-aaf9-9ec214e1e3c4)

**Comment by tnodir** [comment]
Let me check this config with proxy..

**Reply by tnodir** [reply]
Allowed rules now block the connection if not matched, but they should just not block.

I'm fixing the logic of allowed rules processing..

**Comment by tnodir** [comment, marked as answer]
With the [v3.15.0](https://github.com/tnodir/fort/releases/tag/v3.15.0):

1. Create Preset Rules for Allowed/Blocked Proxy:
![proxy-presets](https://github.com/user-attachments/assets/b1cec496-3599-4b62-b4e2-7b7338b7838d)
![proxy-allow](https://github.com/user-attachments/assets/6496a8ec-4557-4071-a7f8-d9b0aef4c7e7)
![proxy-block](https://github.com/user-attachments/assets/185722d1-1c86-474c-8c23-0e6f95676102)

2. Create App Rule for Allowed Proxy:
![proxy-app](https://github.com/user-attachments/assets/2436a07a-ae7b-40e1-8924-43da20349549)

3. Create Global Rule (applied AFTER App Rules) for Blocked Proxy:
![global-post-blocked](https://github.com/user-attachments/assets/8c3ea829-669c-4b31-ac98-f3386a119d77)

4. Set the "Proxy" App Rule to desired Program.

5. Make sure that in the Options:
    - "Filter Local Addresses (127.0.0.0/8 ...)" is turned on
    - "Local Network addresses" does not contain the "127.0.0.1" address

**Reply by EsuRt** [reply]
After following the above tutorial, the issue persists, and it also causes internet-enabled programs to be blocked. Is it possible to configure the settings like in Comodo Firewall? Comodo Firewall can achieve my goal with relatively simple configuration, but it feels large-scale.

Here is the Comodo Firewall setup guide:https://help.comodo.com/topic-399-1-904-11870-application-rules.html

![1](https://github.com/user-attachments/assets/ec281e94-690e-4e2b-b564-1f74da96f317)

![2](https://github.com/user-attachments/assets/9a14053b-4f06-4b95-b511-fa4578d0336c)

![3](https://github.com/user-attachments/assets/8c79dbb6-0741-4ebe-98c3-c43747a3c787)

![4](https://github.com/user-attachments/assets/d3964500-b21a-48a5-af2e-b247656b8c95)

![5](https://github.com/user-attachments/assets/5cb6ca0f-f255-49bc-8af3-2a80441cddab)

![6](https://github.com/user-attachments/assets/12aa66c0-2c68-475d-be0c-ff4606f487a2)

![7](https://github.com/user-attachments/assets/57145c46-353a-4806-a857-affcc1ac1378)

**Reply by tnodir** [reply]
@EsuRt From your screenshot of Blocked connections:
![1](https://github.com/user-attachments/assets/ec281e94-690e-4e2b-b564-1f74da96f317)

- "BarTender" is blocked, because it has "Block Internet traffic" flag checked on.
    (The "127.0.0.1" is Internet address, because it is filtered by "Filter Local Addresses (127...)" and not in the "Local Network addresses" field.)

- Other programs are blocked, because "Global Post" is blocking them by default.
    Set the "Proxy" rule to them (4. of above tutorial) to allow Internet access via proxy.

So, the rules work as requested:
```
I have installed proxy software on my system, with local proxy ports set to 10800 and 10801. I want to prevent software from accessing the internet through these proxy ports but still allow them to access other local ports.
```
I.e. only programs with specified "Proxy" rule can access Internet via the proxy.

**Reply by tnodir** [reply]
@EsuRt Is it working for you now?

**Reply by EsuRt** [reply]
> @EsuRt Is it working for you now?

Due to the existence of Global Post rules, this setup may not meet my requirements at the moment. I have temporarily blocked their internet access entirely without distinguishing between ports to avoid affecting the normal internet access of other applications.

**Reply by tnodir** [reply]
> Due to the existence of Global Post rules, this setup may not meet my requirements at the moment.

Lets start again from requirements:
- you have a local proxy
- all programs should access Internet via proxy or only specified ones?

**Reply by EsuRt** [reply]
> > Due to the existence of Global Post rules, this setup may not meet my requirements at the moment.
> 
> Lets start again from requirements:
> 
> * you have a local proxy
> * all programs should access Internet via proxy or only specified ones?

Here is my specific use case scenario:
For example, I have a local proxy with proxy ports 10800 and 10801, and there are four software applications on the system: A, B, C, and D.
* Software A and D by default access the internet through the proxy.
* Software B and C by default connect to the internet directly without using the proxy.

Now, I want to block software A and B from accessing the internet, but not block software C and D. Additionally, software A will start subprocesses that communicate with the main process through other ports, where subprocess 1 may connect directly to the internet or use the proxy.
To ensure that software A works properly, I need to block access to ports 10800 and 10801 for software A.

**Reply by tnodir** [reply]
> Software A and D by default access the internet through the proxy.

And you do not care, that some other Z program can access the Internet via proxy?

**Reply by tnodir** [reply]
> I want to block software A and B from accessing the internet

Can you just set A and B programs as `Blocked`?

**Reply by EsuRt** [reply]
> > Software A and D by default access the internet through the proxy.
> 
> And you do not care, that some other Z program can access the Internet via proxy?

Yes, I allow some other Z program to access the Internet via proxy.

**Reply by tnodir** [reply]
OK, lets create same rules as in Comodo Firewall:

![5](https://github.com/user-attachments/assets/0a983f40-8d11-402c-a8d6-542b651ddbfd)

1. Create the following Rules in Fort:
![rules](https://github.com/user-attachments/assets/b5d673f0-efb7-4409-9f39-7094a26ac1db)
![allow-non-proxy](https://github.com/user-attachments/assets/c8a69bdb-3a9f-4b66-a5c0-166ce4929768)
![block-non-localhost](https://github.com/user-attachments/assets/1ec7a467-200a-4649-a318-877f3679332f)
![block-proxy](https://github.com/user-attachments/assets/38759be0-2aae-470b-8f6b-2a170119398f)
![block-app-proxy](https://github.com/user-attachments/assets/ce94d315-4536-49d3-be9d-1b26472d1abc)

2. Set the "Block App Proxy" rule to A and B programs

**Reply by EsuRt** [reply]
The new rules can solve my problem. Thank you for your patient guidance.

**Comment by tnodir** [comment]
> Is it possible to configure the settings like in Comodo Firewall?

@EsuRt Can you please show by screenshots, how it looks in Comodo Firewall:
- allow only some programs to access Internet via proxy with `127.0.0.1:TCP(10800-10801)`
- block all other programs by default to access Internet via proxy with `127.0.0.1:TCP(10800-10801)`
- and all programs should still have access to other ports and protocols on `127.0.0.1`

**Reply by EsuRt** [reply]
Comodo Firewall allows individual rules to be set for each program. The following rules can achieve the aforementioned requirements.
![3](https://github.com/user-attachments/assets/bbba2efc-9c9a-4951-964f-162572d398bf)

![4](https://github.com/user-attachments/assets/eb8b8d9d-a496-42aa-b0a7-0ac2260ab965)

![5](https://github.com/user-attachments/assets/0a983f40-8d11-402c-a8d6-542b651ddbfd)

![6](https://github.com/user-attachments/assets/83800ac9-c912-4a5f-9d15-75388579ba9a)

![7](https://github.com/user-attachments/assets/58d00b9c-dcdc-4eaf-9ed5-30e2602df29f)

![8](https://github.com/user-attachments/assets/646eeff9-47e8-48dd-8c20-518ebe46fa79)

![1](https://github.com/user-attachments/assets/6c770569-5b35-42db-a044-320db064a20d)

![2](https://github.com/user-attachments/assets/5d6b5c32-ecff-4386-abef-97d50a34e083)

The optional settings for IP addresses and ports are as follows:
![9](https://github.com/user-attachments/assets/f7d951e1-58db-49d9-b7e9-cdb0c461466b)

![10](https://github.com/user-attachments/assets/35ac9a80-acb5-4890-9ba7-6c287464a91a)

I have tested many network control programs, and currently, only Comodo Firewall and Fort can block network access through folders and filter loopback addresses. Fort is excellent and lightweight, but its port control features are slightly inferior to Comodo Firewall. I hope it can be improved.

**Reply by tnodir** [reply]
Fort's App Rules do not filter the traffic without associated Program. So, you have to specify the App Rule to some Program to have effect.

Only Global Rules filter the traffic themselves.
